### PR TITLE
Detected and fixed an issue related to -0

### PR DESCRIPTION
**Pull request by dubzzz**
I detected and fixed an issue with -0 by using property based tests.

- first commit adds the fix and test for the value -0
- second commit adds the test that discovered the issue.

### CODE REVIEW

File: javascript-stringify.js [hunk 1]
```diff
@@ -214,7 +214,10 @@
     'string': function (string) {
       return "'" + string.replace(ESCAPABLE, escapeChar) + "'";
     },
-    'number': String,
+    'number': function (val) {
+      if (val === 0 && 1 / val === -Infinity) return '-0';
```
Comment (blakeembrey) [inline javascript-stringify.js]: What about using `Object.is`? I realize this has much better compatibility though, so LGTM!

File: test.js [hunk 2]
```diff
@@ -331,4 +334,21 @@ describe('javascript-stringify', function () {
       test(obj, '{a:{b:{}}}', null, { maxDepth: 2, references: true })
     );
   });
+  describe('property based', function () {
+    var customEval = function (repr) {
```
Comment (blakeembrey) [inline test.js]: Would it make sense to just use `eval` for now? I'm not sure we need the additional wrapper.

File: test.js [hunk 3]
```diff
@@ -331,4 +334,21 @@ describe('javascript-stringify', function () {
       test(obj, '{a:{b:{}}}', null, { maxDepth: 2, references: true })
     );
   });
+  describe('property based', function () {
```
Comment (blakeembrey) [inline test.js]: Whitespace above.

**Comment by blakeembrey** [comment]
I love the idea of `fast-check`, I'm so glad someone other than me got around to this finally! I'll look more into it and see how else I can better utilize it to test, especially around some of the more complex code possible (e.g. regexps and properties on objects).

File: javascript-stringify.js (hunk 1 above)
Comment (dubzzz) [reply javascript-stringify.js]: Indeed, I believed `Object.is` was not part of Node 4, but it is.

File: test.js (hunk 2 above)
Comment (dubzzz) [reply test.js]: I think I had an issue with eval, but I will try again ;)

Comment (dubzzz) [reply test.js]: Calling `console.log(eval("{}"))` prints `undefined`.
Calling ` console.log(eval("(function () { return {}; })()"));` does work.

So I can either have:

```js
var customEval = function (repr) {
  return eval("(function () { return " + repr + "; })()");
};
```

or

```js
var customEval = function (repr) {
  return Function("return " + repr)();
};
```

Comment (blakeembrey) [reply test.js]: That makes sense, since it's a block. All good 👍

**Comment by dubzzz** [comment]
Concerning the generation of regex, I wrote a snippet doing that few days ago: https://runkit.com/dubzzz/valid-regex-generator

It is still incomplete but it is a good start ^^

I will maybe include such generator in fast-check ;)

**Pull request merged**